libplatsupport/plat_include/star64/platsupport/plat/timer.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2023, UNSW
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/*
+ * The JH7110 SoC contains a timer with four 32-bit counters. Each one of these
+ * counters is referred to as a "channel".
+ */
+
+/* Channel information */
+#define STARFIVE_TIMER_NUM_CHANNELS 4
+#define STARFIVE_TIMER_CHANNEL_REGISTERS_LEN_IN_BYTES 0x40
+#define STARFIVE_TIMER_CHANNEL_0_IRQ 69
+#define STARFIVE_TIMER_CHANNEL_1_IRQ 70
+#define STARFIVE_TIMER_CHANNEL_2_IRQ 71
+#define STARFIVE_TIMER_CHANNEL_3_IRQ 72
+
+/* This information comes from the DTS file */
+#define STARFIVE_TIMER_BASE 0x13050000
+#define STARFIVE_TIMER_REGISTER_WINDOW_LEN_IN_BYTES 0x10000
+#define STARFIVE_TIMER_TICKS_PER_SECOND 0x16e3600
+
+#define STARFIVE_TIMER_MAX_TICKS 0xffffffff
```

Comment:
Do we have `UINT32_MAX` here?
```suggestion
#define STARFIVE_TIMER_MAX_TICKS  UINT32_MAX
```